refactor(docs-infra): clean up CSS styles

gkalpak opened the pull request:
This PR contains several refactoring commits that clean up our angular.io styles. All commits are refactorings (i.e. no feature/fix), so there should be no visual difference in the built app. (If there is, that is a mistake 😱)

See the individual commits for details. 

NOTES:
- Most commits are easier to review with whitespace changes ignored. (In GitHub, this can be achieved by appending `?w=1` to the URL.)
- I have tried to limit changes to a signle file/component in each commit (to keep them easier to review).
- There are three main types of clean-ups:
    - Removing unused styles.
    - Merging blocks that target the same elements.
    - Scope styles to specific components/elements.

mary-poppins commented:
You can preview 92f324a at https://pr40704-92f324a.ngbuilds.io/.

petebacondarwin reviewed (approved):
Nice tidy up. A couple of comments but otherwise LGTM.

gkalpak commented on aio/src/styles/1-layouts/_sidenav.scss:
```diff
@@ -73,119 +55,139 @@ mat-sidenav-container.sidenav-container {
-    width: 24px;
-  }
-}
+  aio-nav-item {
```
It actually expands to `aio-nav-menu aio-nav-item .vertical-menu-item`. Are you suggesting to change it to `aio-nav-menu .vertical-menu-item`?

petebacondarwin replied:
My mistake. Carry on

gkalpak committed:
refactor(docs-infra): remove unused styles from `_content-layout.scss`

This commit removes some styles from `_content-layout.scss` that are no
longer used (i.e. either their CSS selectors do not match any element in
the app or they are overridden by other rules).

gkalpak committed:
refactor(docs-infra): remove unused styles from `_footer.scss`

This commit removes some styles from `_footer.scss` that are no longer
used (i.e. either their CSS selectors do not match any element in the
app or they are overridden by other rules).

gkalpak committed:
refactor(docs-infra): scope footer styles to the `<aio-footer>` element

This commit ensures that all styles for the page footer (which are
defined in `_footer.scss`) only apply to elements inside an
`<aio-footer>` element. This will prevent the styles accidentally taking
effect on a different part of the app.

gkalpak committed:
refactor(docs-infra): merge style rules in `_footer.scss`

This commit cleans up the styles in `_footer.scss` by merging together
blocks that target the same elements.

gkalpak committed:
refactor(docs-infra): remove unused styles from `_layout-global.scss`

This commit removes some styles from `_layout-global.scss` that are no
longer used (i.e. either their CSS selectors do not match any element in
the app or they are overridden by other rules).

gkalpak committed:
refactor(docs-infra): remove unused styles from `_sidenav.scss`

This commit removes some styles from `_sidenav.scss` that are no longer
used (i.e. either their CSS selectors do not match any element in the
app or they are overridden by other rules).

gkalpak committed:
refactor(docs-infra): scope nav-item styles to the `<aio-nav-item>` element

This commit ensures that all styles for the sidenav nav-items (which are
defined in `_sidenav.scss`) only apply to elements inside an
`<aio-nav-item>` element. This will prevent the styles accidentally
taking effect on a different part of the app.

gkalpak committed:
refactor(docs-infra): merge style rules in `_sidenav.scss`

This commit cleans up the styles in `_sidenav.scss` by merging
together blocks that target the same elements.

It also applies the equivalent changes in `_notification.scss` rules
that override sidenav styles.

gkalpak committed:
refactor(docs-infra): merge style rules in `_top-menu.scss`

This commit cleans up the styles in `_top-menu.scss` by merging
together blocks that target the same elements.

It also makes similar changes to selectors in `_notification.scss` rules
that override top-menu styles (for consistency).

gkalpak committed:
refactor(docs-infra): scope top-menu styles to the `.app-toolbar` element

This commit ensures that all styles for the app top-menu (which are
defined in `_top-menu.scss`) only apply to elements inside an
`.app-toolbar` element. This will prevent the styles accidentally taking
effect on a different part of the app.

gkalpak committed:
refactor(docs-infra): move common `.group-buttons` styles to `_buttons.scss`

Both `<aio-contributor-list>` and `<aio-resource-list>` contain
`.group-buttons` elements. Previously, global styles for
`.group-buttons` were defined in `_contributor.scss` (i.e. the styles
from `_contributor.scss` were also applied to `.group-buttons` in
`<aio-resource-list>`). Also, `_contributor.scss` and `_resources.scss`
defined some duplicate styles for `.group-buttons`.

This commit moves the `.group-buttons` styles to `_buttons.scss`, which
contains common button-related styles.

gkalpak committed:
refactor(docs-infra): remove unused styles from `_code.scss`

This commit removes some styles from `_code.scss` that are no longer
used (i.e. either their CSS selectors do not match any element in the
app or they are overridden by other rules).

gkalpak committed:
refactor(docs-infra): scope code example styles to the `<aio-code>` element

This commit ensures that all styles for the code examples (which are
defined in `_code.scss`) only apply to elements inside an `<aio-code>`
element. This will prevent the styles accidentally taking effect on a
different part of the app.

gkalpak committed:
refactor(docs-infra): merge style rules in `_code.scss`

This commit cleans up the styles in `_code.scss` by merging together
blocks that target the same elements.

It also normalizes the rules targeting Angular Material tabs to use
consistent selectors and use class names instead of tag names.

gkalpak committed:
refactor(docs-infra): merge style rules in `_details.scss`

This commit cleans up the styles in `_details.scss` by merging
together blocks that target the same elements.

It also changes the `summary` selector to `details > summary`, but that
should not make any difference in practice, since `<summary>` elements
always appear as direct childs of `<details>`.

gkalpak committed:
refactor(docs-infra): scope contributor-list styles to the `<aio-contributor-list>` element

This commit ensures that all styles for the contributor list (which are
defined in `_contributor.scss`) only apply to elements inside an
`<aio-contributor-list>` element. This will prevent the styles
accidentally taking effect on a different part of the app.

gkalpak committed:
refactor(docs-infra): remove unused styles from `_contributor.scss`

This commit removes some styles from `_contributor.scss` that are no
longer used (i.e. either their CSS selectors do not match any element in
the app or they are overridden by other rules).

gkalpak committed:
refactor(docs-infra): remove unused `_edit-page-cta.scss` file

This commit removes the `_edit-page-cta.scss` file, which is no longer
used (i.e. the rules it contains do not match any element in the app).

gkalpak committed:
refactor(docs-infra): fix indentation in `_label.scss`

This commit makes the indentation in `_label.scss` consistent with that
in most other `.scss` files in `aio/`.

gkalpak committed:
refactor(docs-infra): remove unused styles from `_label.scss`

This commit removes some styles from `_label.scss` that are no longer
used (i.e. their CSS selectors do not match any element in the app).

gkalpak committed:
refactor(docs-infra): merge style rules in `_label.scss`

This commit cleans up the styles in `_label.scss` by merging together
blocks that target the same elements.

gkalpak committed:
refactor(docs-infra): merge style rules in `_presskit.scss`

This commit cleans up the styles in `_presskit.scss` by merging together
blocks that target the same elements.

gkalpak committed:
refactor(docs-infra): remove unused styles from `_resources.scss`

This commit removes some styles from `_resources.scss` that are no
longer used (i.e. either their CSS selectors do not match any element in
the app or they are overridden by other rules).

gkalpak committed:
refactor(docs-infra): merge style rules in `_resources.scss`

This commit cleans up the styles in `_resources.scss` by merging
together blocks that target the same elements.

gkalpak committed:
refactor(docs-infra): scope Resources styles to the `<aio-resource-list>` element

This commit ensures that all styles for the "Resources" page (which are
defined in `_resources.scss`) only apply to elements inside an
`<aio-resource-list>` element. This will prevent the styles accidentally
taking effect on a different part of the app.

gkalpak committed:
refactor(docs-infra): merge style rules in `_toc.scss`

This commit cleans up the styles in `_toc.scss` by merging together
blocks that target the same elements.

gkalpak committed:
refactor(docs-infra): scope select-menu styles to the `<aio-select>` element

This commit ensures that all styles for the select-menu element (which
are defined in `_select-menu.scss`) only apply to elements inside an
`<aio-select>` element. This will prevent the styles accidentally taking
effect on a different part of the app.

gkalpak committed:
refactor(docs-infra): scope TOC styles to the `<aio-toc>` element

This commit ensures that all styles for the TOC (which are defined in
`_toc.scss`) only apply to elements inside an `<aio-toc>` element. This
will prevent the styles accidentally taking effect on a different part
of the app.

gkalpak committed:
refactor(docs-infra): remove unused styles from `_api-list.scss`

This commit removes some styles from `_api-list.scss` that are no longer
used (i.e. either their CSS selectors do not match any element in the
app or they are overridden by other rules).

gkalpak committed:
refactor(docs-infra): move common `.symbol` styles to `_api-symbols.scss`

The `.symbol` CSS class (which is primarily used to denote the type of
API symbols - classes, functions, interfaces, etc.) are used in several
places:
- In the API list page (with the corresponding styles defined in
  `_api-list.scss`).
- In search results (with the corresponding styles defined in
  `_errors.scss`).
- In error list page (with the corresponding styles defined in
  `search-results.scss`).
- In the `<aio-select>` component (with the corresponding styles defined
  in `_select-menu.scss`).

Previously, global styles for `.symbol` were defined in `_api-list.scss`
(i.e. the styles from `_api-list.scss` were also applied to `.symbol`
elements in other places/components). Also, some of the SCSS files
mentioned above defined some duplicate styles for `.symbol`.

This commit moves the `.symbol` styles to a new `_api-symbols.scss`
file, which contains common symbol-related styles.

gkalpak committed:
refactor(docs-infra): merge style rules in `_api-list.scss`

This commit cleans up the styles in `_api-list.scss` by merging together
blocks that target the same elements.

gkalpak committed:
refactor(docs-infra): move common `.code-anchor` styles to `_code.scss`

The `.code-anchor` class can be used anywhere where we have code
examples (including API pages and docs guides). Previously, global
styles for `.code-anchor` were defined in `_api-list.scss` (i.e. the
styles from `_api-list.scss` were also applied to `.code-anchor`
elements in other pages/components).

This commit moves the `.code-anchor` styles to `_code.scss`, which
contains other common code-related styles.

gkalpak committed:
refactor(docs-infra): use a single CSS class to target the API list container

Previously, the API list container (in the template of the
`<aio-api-list>` component) had three different CSS classes
(`.api-list-container`, `.docs-content`, `.l-content-small`) that were
all used for styling it. This seemed unnecessary and made it more difficult
to see what styles were applied to the container.

This commit removes the extra classes and consolidates the styles under
the `.api-list-container` class (which was the most descriptive one).

gkalpak committed:
refactor(docs-infra): scope API list styles to the `<aio-api-list>` element

This commit ensures that all styles for the API list page (which are
defined in `_api-list.scss`) only apply to elements inside an
`<aio-api-list>` element. This will prevent the styles accidentally
taking effect on a different part of the app.

gkalpak committed:
fixup! refactor(docs-infra): remove unused styles from `_sidenav.scss`

gkalpak commented on aio/src/styles/1-layouts/_sidenav.scss:
```diff
@@ -45,15 +32,45 @@ mat-sidenav-container.sidenav-container {
-  background-color: $white;
-}
+  .sidenav-content {
+    background-color: $white;
```
Removed.

mary-poppins commented:
You can preview 0a1301b at https://pr40704-0a1301b.ngbuilds.io/.
